dask/multiprocessing.py

```diff
@@ -191,10 +195,18 @@ def get(
             os.environ["PYTHONHASHSEED"] = "6640"
         context = get_context()
         pool = ProcessPoolExecutor(
-            num_workers, mp_context=context, initializer=initialize_worker_process
+            num_workers,
+            mp_context=context,
+            initializer=initializer or initialize_worker_process,
```

Comment:
Should we have this replace the initializer or should we have it be called within the initializer function?

IOW do we want to override the initializer's behavior or append to it? Reasonable options either way. Just worth evaluating both

Reply:
Indeed.  I also wondered about this.  I suppose the default initializer `initialize_worker_process()` is important enough to justify remaining permanent.  Though I do not know exactly why.

Anyway, I've now changed the default behavior to *append* the default initializer with the user initializer function (if given), while providing an option (`override_default_initializer=True`) to easily override the default initializer if desired.

Reply:
Yeah this seems reasonable generally. Had one question below about using the config

The seed initialization code has a lot of history. PR ( https://github.com/dask/distributed/pull/835 ) discovered the issue, which specifically is that NumPy doesn't regenerate the seed after forking. There was some work done in CPython, which NumPy could leverage though to do this regeneration so we don't have to. Raised issue ( https://github.com/numpy/numpy/issues/21528 ) about that.

There is a different issue around hash seed handling that comes up in issue ( https://github.com/dask/dask/issues/6640 ), which is also being handled here.

In any event, think the key takeaway is Dask periodically learns these unusual edge cases and handles them for users internally as part of the initialization of worker processes. There's some value in keeping those learnings when changing out the initializer. Though if someone really wants full control, it could make sense to replace the initializer altogether. IOW what is now being done here 🙂

Reply:
I'll leave this conversation unresolved to allow other readers to see your answer, since it is quite informative.

Reply:
Sounds good. Marked the rest of the threads resolved. Though feel to change others if needed

Reply:
This is also a good point @jcrist .  I'll wait a bit more to see if there is some consensus about this.

Reply:
Yeah, I agree with @jcrist that always running the default initializer and removing `use_default_initializer` seems like a good move

Reply:
Given the PR started by not running the default initializer, at least having the option to enable/disable seemed better. Though if there is no need to disable, guess it makes sense to drop the option and always run the default initializer

Reply:
It seems there is general agreement on dropping the `use_default_initializer` option.  So I'll do that promptly.  Thanks to all for making your views known!